src/promptflow/promptflow/batch/batch_engine.py

```diff
@@ -0,0 +1,189 @@
+import asyncio
+import uuid
+from pathlib import Path
+from typing import Any, Dict, List, Mapping, Optional
+
+from promptflow._constants import LINE_NUMBER_KEY
+from promptflow._core._errors import UnexpectedError
+from promptflow._utils.context_utils import _change_working_dir
+from promptflow._utils.execution_utils import (
+    apply_default_value_for_input,
+    collect_lines,
+    get_aggregation_inputs_properties,
+    handle_line_failures,
+)
+from promptflow._utils.logger_utils import logger
+from promptflow._utils.utils import dump_list_to_jsonl, resolve_dir_to_absolute, transpose
+from promptflow.batch._batch_inputs_processor import BatchInputsProcessor
+from promptflow.batch.base_executor_proxy import AbstractExecutorProxy
+from promptflow.batch.python_executor_proxy import PythonExecutorProxy
+from promptflow.contracts.flow import Flow, ProgramLanguage
+from promptflow.contracts.run_info import Status
+from promptflow.exceptions import PromptflowException
+from promptflow.executor._result import AggregationResult, BulkResult, LineResult
+from promptflow.executor.flow_validator import FlowValidator
+from promptflow.storage._run_storage import AbstractRunStorage
+
+OUTPUT_FILE_NAME = "output.jsonl"
+
+
+class BatchEngine:
+    """This class is used to execute flows in batch mode"""
+
+    executor_proxy_classes: Mapping[str, AbstractExecutorProxy] = {
+        ProgramLanguage.PYTHON: PythonExecutorProxy,
+    }
+
+    @classmethod
+    def register_executor(cls, type: ProgramLanguage, executor_proxy_cls: AbstractExecutorProxy):
```

Comment:
Remember to add docstring for public interfaces

Reply:
Added.